src/boot/bootctl-install.c

```diff
@@ -449,7 +455,156 @@ static int copy_one_file(const char *esp_path, const char *name, bool force) {
+
+        /* parse only the first boot.csv we find if requested */
+        if (!args->loader_path || !args->loader_description
+                        || *args->loader_path || *args->loader_description
+                        || !endswith_no_case(path, "boot.csv") || (r < 0 && !IN_SET(r, -ESTALE, -ESRCH)))
+                return RECURSE_DIR_CONTINUE;
+
+        _cleanup_free_ char *csv = NULL;
+        size_t csv_size = 0;
+        r = read_full_file(source_path, &csv, &csv_size);
+        if (r < 0) {
+                log_error_errno(r, "Failed to read %s: %m", path);
+                return RECURSE_DIR_CONTINUE;
+        }
+
+        _cleanup_free_ char *file = NULL, *desc = NULL;
+        r = parse_boot_csv_mem((char16_t*)csv, csv_size, &file, &desc);
+        if (r < 0)
+                return RECURSE_DIR_CONTINUE;
+
+        _cleanup_free_ char *dir = NULL;
+        if (path_extract_directory(path, &dir) < 0)
+                return RECURSE_DIR_CONTINUE;
+
+        _cleanup_free_ char *p = path_join(dir, file);
+        if (!p)
+                return log_oom();
+
+        log_debug("Found default loader %s in %s", p, source_path);
+        *args->loader_path = TAKE_PTR(p);
+        *args->loader_description = TAKE_PTR(desc);
+
+        return RECURSE_DIR_CONTINUE;
+}
+
+static int install_binaries_extra(const char *esp_path, const char *arch, bool force,
+                char** ret_loader_path, char** ret_loader_description) {
+        char *root = IN_SET(arg_install_source, ARG_INSTALL_SOURCE_AUTO, ARG_INSTALL_SOURCE_IMAGE) ? arg_root : NULL;
+        _cleanup_closedir_ DIR *d = NULL;
+        _cleanup_free_ char *path = NULL, *loader_path = NULL, *loader_description = NULL;
+        int r;
+
+        r = chase_and_opendir(BOOTLIBDIR "/extra", root, CHASE_PREFIX_ROOT|CHASE_PROHIBIT_SYMLINKS, &path, &d);
+        /* If we had a root directory to try, we didn't find it and we are in auto mode, retry on the host */
+        if (r == -ENOENT && root && arg_install_source == ARG_INSTALL_SOURCE_AUTO)
+                r = chase_and_opendir(BOOTLIBDIR "/extra", NULL, CHASE_PREFIX_ROOT|CHASE_PROHIBIT_SYMLINKS, &path, &d);
+        if (r == -ENOENT)
+                return 0; /* ignore if vendor dir does not exist */
+        if (r < 0)
+                return log_error_errno(r, "Failed to open " BOOTLIBDIR "/extra: %m");
+
+        r = recurse_dir(dirfd(d), NULL, 0, 3, RECURSE_DIR_IGNORE_DOT, copy_loader,
+                        &(struct CopyLoaderArgs) {
+                                .source_path = BOOTLIBDIR "/extra",
+                                .esp_path = esp_path,
+                                .force = force,
+                                .loader_path = &loader_path,
+                                .loader_description = &loader_description,
+                        });
+        if (r < 0)
+                return log_error_errno(r, "Failed to recurse %s: %m", path);
+
+        if (ret_loader_path && loader_path)
```

Comment:
we have this rule in our tree: always initialize all return parameters on success + never touch them on failure. hence, please initialize *ret_loader_path here given you return successfuly. i.e. drop the `&& loader_path`